### PR TITLE
chore: add tests for `solve` in Datalog

### PR DESCRIPTION
**Pull request by JumpTwiice**
Related to https://github.com/flix/flix/issues/11065.

### CODE REVIEW

File: main/test/flix/Test.Exp.Fixpoint.KeywordSolve.flix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            ASolve01(42).
+        };
+        let otherProg = #{
+            ASolve01(-42).
+        };
+        let k = solve solved, otherProg;
+        let res1 = query solved select x from ASolve01(x);
+        let res2 = query k select x from ASolve01(x);
+        res1 `Assert.eq` Vector#{42} and res2 `Assert.eq` Vector#{-42, 42}
+    }
```
Comment (magnus-madsen) [inline main/test/flix/Test.Exp.Fixpoint.KeywordSolve.flix:36]: I think these tests would be easier to read if it was simply:

db1
db2

and/or mm1 and mm2.

That way the text aligns better.

I would also just call the predicate `A`.

File: main/test/flix/Test.Exp.Fixpoint.KeywordSolve.flix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        };
+        let otherProg = #{
+            ASolve03(-x) :- BSolve03(x).
+        };
+        let k = solve solved, otherProg;
+        let res1 = query solved select x from ASolve03(x);
+        let res2 = query k select x from ASolve03(x);
+        res1 `Assert.eq` Vector#{1, 2} and res2 `Assert.eq` Vector#{-2, -1, 1, 2}
+    }
+
```
Comment (magnus-madsen) [inline main/test/flix/Test.Exp.Fixpoint.KeywordSolve.flix:92]: I feel like we are missing some tests that facts from one program does not leak into another. Abstractly:

db1 = #{A(1)}
db2 = #(B(1)}
solve db1, db2
solve db1 // should not contain B fact
solve db2 // should not contain A fact

File: main/test/flix/Test.Exp.Fixpoint.KeywordSolve.flix (hunk 1 above)
Comment (JumpTwiice) [reply main/test/flix/Test.Exp.Fixpoint.KeywordSolve.flix:36]: The issue with calling the predicate `A` is that it might not stratify if another test/program uses the same predicate in a negative atom.

Comment (magnus-madsen) [reply main/test/flix/Test.Exp.Fixpoint.KeywordSolve.flix:36]: I see

Comment (magnus-madsen) [reply main/test/flix/Test.Exp.Fixpoint.KeywordSolve.flix:36]: But I think tests are independent, so only within the same file.

**Review by magnus-madsen — APPROVED** [review]
